spatialdata/_core/_spatialdata.py

```diff
@@ -0,0 +1,262 @@
+
+        if table is not None:
+            Table_s.validate(table)
+            self._table = table
+
+    def write(self, file_path: str) -> None:
+        """Write to Zarr file."""
+
+        store = parse_url(file_path, mode="w").store
+        root = zarr.group(store=store)
+
+        # get union of unique ids of all elements
+        elems = set().union(
+            *[
+                set(i)
+                for i in [
+                    self.images,
+                    self.labels,
+                    self.points,
+                    self.polygons,
+                    self.shapes,
+                ]
+            ]
+        )
+
+        for el in elems:
+            elem_group = root.create_group(name=el)
+            if self.images is not None and el in self.images.keys():
+                write_image(
+                    image=self.images[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.labels is not None and el in self.labels.keys():
+                write_labels(
+                    labels=self.labels[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.polygons is not None and el in self.polygons.keys():
+                write_polygons(
+                    polygons=self.polygons[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.shapes is not None and el in self.shapes.keys():
+                write_shapes(
+                    shapes=self.shapes[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.points is not None and el in self.points.keys():
+                write_points(
+                    points=self.points[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+
+        if self.table is not None:
+            write_table(table=self.table, group=root, name="table")
+
+    @property
+    def table(self) -> AnnData:
+        return self._table
+
+    @classmethod
+    def read(cls, file_path: str) -> SpatialData:
+        from spatialdata._io.read import read_zarr
+
+        sdata = read_zarr(file_path)
+        return sdata
+
+    def __repr__(self) -> str:
+        return self._gen_repr()
+
+    def _gen_repr(
+        self,
+    ) -> str:
+        def rreplace(s: str, old: str, new: str, occurrence: int) -> str:
+            li = s.rsplit(old, occurrence)
+            return new.join(li)
+
+        def h(s: str) -> str:
+            return s
+            # return hashlib.md5(repr(s).encode()).hexdigest()
+
+        ##
```

Comment:
Done it (on the code I will push). I removed `return s` in favor of the hash string (this implements the logic to print nested things by later replacing the hash with the correct graphics elements). Using `s` instead of the hash risks collision with an unfortunate choice of names from the user.

Reply:
the `##` are for [PyCharm cell mode](https://plugins.jetbrains.com/plugin/7858-pycharm-cell-mode), I use them while  debugging/coding. Feel free to remove them.